src/system-health/health_checker/hardware_checker.py

```diff
@@ -133,6 +134,18 @@ def _check_fan_status(self, config):
                                                    speed_tolerance))
                         continue
 
+            if not self._ignore_check(config.ignore_devices, 'fan', name, 'direction'):
+                direction = data_dict.get('direction', 'N/A')
+                # ignore fan whose direction is not available to avoid too many false alarms
+                if direction != 'N/A':
+                    if not expect_fan_direction:
+                        # initialize the expect fan direction
+                        expect_fan_direction = (name, direction)
+                    elif direction != expect_fan_direction[1]:
```

Comment:
@Junchao-Mellanox I don't understand why `expect_fan_direction` depends upon the `direction` value against which it is checking?

Reply:
Hi, it is depending on the first fan's direction.